File: 프로그래머스/lv1/12934. 정수 제곱근 판별/정수 제곱근 판별.java
class Solution {
    public long solution(long n) {
        double x = Math.pow(n, 0.5);
        double answer = 0;
        
        if(x%1 != 0) {
            answer = -1;    
        }else {
            answer = Math.pow(x+1,2);
        }
        return (long)answer;
    }
}

File: 프로그래머스/1/12934. 정수 제곱근 판별/정수 제곱근 판별.go
func solution(n int64) int64 {
    if n == 1 {
        return 4
    }
    
    if n < 3 {
        return -1
    }
    
    var i int64
    for i = 2 ; i*i <= n; i++ {
        if i*i == n {
            return (i+1) * (i+1)
        }
    }
    
    return -1
}